client/types.go

```diff
@@ -427,7 +427,11 @@ type Signup struct {
 }
 
 type Tenant struct {
-	Signup Signup `json:"signup"`
+	Signup              Signup  `json:"signup"`
+	FromEmail           *string `json:"fromEmail,omitempty"`
+	FinanceSupportEmail *string `json:"financeSupportEmail,omitempty"`
+	SiteAccessCode      *string `json:"siteAccessCode,omitempty"`
```

Comment:
This is the API response when you request tenant to the 3scale API
```json
{
  "signup": {
    "account": {
      "id": 2,
      "state": "approved",
      "org_name": "Provider Name",
      "support_email": "admin@3scale.eguzki.apps.dev-eng-ocp4.dev.3sca.net",
      "admin_domain": "3scale-admin.eguzki.apps.dev-eng-ocp4.dev.3sca.net",
      "domain": "3scale.eguzki.apps.dev-eng-ocp4.dev.3sca.net"
    },
    "access_token": {
      "id": 0,
      "name": "",
      "scopes": null,
      "permission": "",
      "value": ""
    }
  }
}
```

I would stick the types to the direct representation of the entities. I would not build types in this repo that gets info from multiple 3scale API endpoints. That's up to the caller of this SDK.

Reply:
Hi @eguzki , thank you very much for comment.

Your example seems for `account 2` that is default tenant, admin, it's not created by Tenant CR, and it should not be impacted by current PR.
Below are fragments of responces for Tenant that created from 3scale Tenant CR.
- Response 1 - Tenant CR does Not contains New Info fields
- Response 2 - Tenant CR contains New Info fields
We can see that default fields that were initially taken from Email parameter - replaced by new fields from new Info specific email paramaters.

**Response 1 Tenant CR does Not contains New Info fields**
support_email and finance_support_email here looks like populated from `Email`

```yaml
    <from_email>no-reply@apps.vmo01.7xu7.s1.devshift.org</from_email>
    <support_email>vmo@kuku.com</support_email>
    <finance_support_email>vmo@kuku.com</finance_support_email>
    <site_access_code>d29cd1312e</site_access_code>
```

**Response 2 Tenant CR contains New Info fields**
- Tenant CR was updated, new Info fields added, and populated in Porta.

```yaml
After addinig Info fields in Tenant CR:
      <from_email>tenant1from@kuku.com</from_email>
      <support_email>tenant1support@kuku.com</support_email>
      <finance_support_email>tenant1fin@kuku.com</finance_support_email>
      <site_access_code>abc123</site_access_code>
```

Not sure, if I answered on your comment. If not, could you please give me more details for implementation. Thank you!

Reply:
cc @MStokluska

Reply:
Which 3scale API endpoint is providing those new fields? `GET /master/api/providers/{id}.json` does not. 

Try to stick to json format as far as you can. `XML` format is being deprecated from 3scale API's AFAIK.

Reply:
full responces added to Task  https://issues.redhat.com/browse/THREESCALE-10672 , for comparing before and after adding new info parameters to Tenant CR

Reply:
you shared the xml endpoint response. The xml endpoints can be very different to json endpoints. I see that the xml endpoint response for the tenant is mixing accounts, users, plans.... 

You plan to be using xml endpoints?

Reply:
> full responces added to Task https://issues.redhat.com/browse/THREESCALE-10672 , for comparing before and after adding new info parameters to Tenant CR

I am talking about 3scale API endpoints. This has nothing to do with the new fields of the Tenant CRD.

Reply:
In other words,

When I call `ShowTenant()` method, that returns a `Tenant` object, are those new fields you added being filled? I would say they are not. But I could be wrong.

Reply:
Master Branch was tested , to compare wih response in PR when No Info field in Tenant CR (same Tenant yaml file used for test) - responses are identical (exept time, and site_access_code)

Reply:
sorry I do not follow. I made a few question and I had no answer:

> Which 3scale API endpoint is providing those new fields?

> You plan to be using xml endpoints?

> When I call ShowTenant() method, that returns a Tenant object, are those new fields you added being filled?

Reply:
cc @MStokluska @eguzki

Reply:
sorry by mistake at morning was placed wrong log for Show tenant (xml instead of json). Now replaced.

Reply:
I see the fields at the `Account` level in the json response. You added at the `Tenant`Level. Thus,no way `ShowTenant()` is going to work as it is now. 

Furthermore, the json tags are wrong and case type does not match the 3scale API response fields. The fields in the tenant type are defined in  camel case, whereas the 3scale API are formatted in snake case.

Reply:
updated. Thank you for comments.
`$ curl -X 'GET' 'https://master.apps.vmo01.7xu7.s1.devshift.org/master/api/providers/4.json?access_token=4YUkqSdl' -H 'accept: */*'
{"signup":{"account":{"id":4,"created_at":"2024-01-24T12:54:18Z","updated_at":"2024-01-24T13:00:58Z","admin_domain":"myorg2-admin.apps.vmo01.7xu7.s1.devshift.org","domain":"myorg2.apps.vmo01.7xu7.s1.devshift.org","admin_base_url":"https://myorg2-admin.apps.vmo01.7xu7.s1.devshift.org","base_url":"https://myorg2.apps.vmo01.7xu7.s1.devshift.org","from_email":"tenant1from@kuku.com","support_email":"tenant1support@kuku.com","finance_support_email":"tenant1fin@kuku.com","site_access_code":"abc123","credit_card_stored":false,"monthly_billing_enabled":true,"monthly_charging_enabled":true,"state":"approved","links":[{"rel":"users","href":"https://master.apps.vmo01.7xu7.s1.devshift.org/admin/api/accounts/4/users"}],"org_name":"myOrg2"}}}[vmogilev@vmogilev THREESCALE-10525]$ `